src/Libraries/Microsoft.Extensions.Http.Telemetry/Logging/Internal/Log.cs

```diff
@@ -26,11 +42,16 @@ public static void OutgoingRequestError(ILogger logger, LogRecord record, Except
         OutgoingRequest(logger, LogLevel.Error, 2, nameof(OutgoingRequestError), record, exception);
     }
 
-    [LogMethod(3, LogLevel.Error, "An error occurred while enriching the log record.")]
-    public static partial void EnrichmentError(ILogger logger, Exception exception);
+    [LogMethod(LogLevel.Error, RequestReadErrorMessage)]
+    public static partial void RequestReadError(ILogger logger, Exception exception, HttpMethod httpMethod, string? httpHost, string httpPath);
+
+    [LogMethod(LogLevel.Error, ResponseReadErrorMessage)]
+    public static partial void ResponseReadError(ILogger logger, Exception exception, HttpMethod httpMethod, string httpHost, string httpPath);
 
-    // Using the code below to avoid every item in ILogger's logRecord State being prefixed with parameter name.
-    // To be fixed in R9 later.
+    [LogMethod(LogLevel.Error, EnrichmentErrorMessage)]
+    public static partial void EnrichmentError(ILogger logger, Exception exception, HttpMethod httpMethod, string httpHost, string httpPath);
+
+    // Using the code below instead of generated logging method because we have a custom formatter and custom tag keys for headers.
```

Comment:
This should be rewritten to use a LoggerMessageState object instead of the LogMethodHelper stuff (which I'm going to delete soon

Reply:
Good catch, will update these methods

Reply:
I don't think it makes sense here, but I want to bring it up anyway.

With the new logger model, redaction is handled inside the logger. Basically, when you have the LoggerMessageState object, you can add normal properties to it (name/value pairs) or you can add classified properties to it (name/value/data class). Based on the data class, the ILogger implementation performs redaction.

In this code, we are redacting the path explicitly within this component, and then calling the logger. Would/could it be simpler to delegate redaction to the logger implementation?

Reply:
Not the whole path segment will be redacted, each path parameter is redacted separately

Reply:
Rewritten to `LoggerMessageState`